src/coreComponents/constitutive/fluid/multifluid/compositional/CompositionalMultiphaseFluid.cpp

```diff
@@ -43,7 +43,7 @@ CompositionalMultiphaseFluid::CompositionalMultiphaseFluid( string const & name,
 
   registerWrapper( viewKeyStruct::equationsOfStateString(), &m_equationsOfState ).
     setInputFlag( InputFlags::REQUIRED ).
-    setDescription( "List of equation of state types for each phase" );
+    setDescription( "List of equation of state types for each phase. Can be PR (= PENG_ROBINSON) or SRK (= REDLICH_KWONG_SOAVE)." );
```

Comment:
I think it is because we need to define template specialization for each enum in enum arrays for these functions:
`PackDevice`,`PackByIndexDevice`,`UnpackDataDevice`,`PackDataDevice`,`UnpackDataByIndexDevice`,`PackDataByIndexDevice`,`UnpackByIndexDevice`,`UnpackDevice`,`PackDataDevice`

(maybe we could implement that in `ENUM_STRINGS()`?)

Here is the [make log.](https://github.com/GEOS-DEV/GEOS/files/13318292/make.log.txt)

Reply:
Ouch. Thanks for checking that. Ideally, I would implement one additional specialization of each of those (un)packing functions that catches all enums (`enable_if<is_enum<T>>`) and delegates to the specialization for the underlying integer type. However this requires the ability to type-cast `ArrayView`s which we are currently missing.